{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running fastp, Kaiju, and Modifying Output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Download, install, and add parallel, fastp and Kaiju to your PATH according to their protocol\n",
    "#### Download and create Kaiju database, place nodes.dmp, names.dmp, and .fmi files inside fastp_Kaiju_processing/input directory"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quality Control:\n",
    "Perform quality control on the reads using fastp and parallel \n",
    "The settings are the default with phred quality >=Q15 \n",
    "This will also output a .html file with information on the quality of the sequences \n",
    "The first line is a dry run that shows what will be executed\n",
    "Second line is without dry run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10_S10_R1_001.fastq.gz\r\n",
      "10_S10_R2_001.fastq.gz\r\n",
      "11_S11_R1_001.fastq.gz\r\n",
      "11_S11_R2_001.fastq.gz\r\n",
      "12_S12_R1_001.fastq.gz\r\n",
      "12_S12_R2_001.fastq.gz\r\n",
      "13_S13_R1_001.fastq.gz\r\n",
      "13_S13_R2_001.fastq.gz\r\n",
      "14_S14_R1_001.fastq.gz\r\n",
      "14_S14_R2_001.fastq.gz\r\n",
      "15_S15_R1_001.fastq.gz\r\n",
      "15_S15_R2_001.fastq.gz\r\n",
      "16_S16_R1_001.fastq.gz\r\n",
      "16_S16_R2_001.fastq.gz\r\n",
      "17_S17_R1_001.fastq.gz\r\n",
      "17_S17_R2_001.fastq.gz\r\n",
      "18_S18_R1_001.fastq.gz\r\n",
      "18_S18_R2_001.fastq.gz\r\n",
      "19_S19_R1_001.fastq.gz\r\n",
      "19_S19_R2_001.fastq.gz\r\n",
      "1_S1_R1_001.fastq.gz\r\n",
      "1_S1_R2_001.fastq.gz\r\n",
      "20_S20_R1_001.fastq.gz\r\n",
      "20_S20_R2_001.fastq.gz\r\n",
      "21_S21_R1_001.fastq.gz\r\n",
      "21_S21_R2_001.fastq.gz\r\n",
      "22_S22_R1_001.fastq.gz\r\n",
      "22_S22_R2_001.fastq.gz\r\n",
      "2_S2_R1_001.fastq.gz\r\n",
      "2_S2_R2_001.fastq.gz\r\n",
      "3_S3_R1_001.fastq.gz\r\n",
      "3_S3_R2_001.fastq.gz\r\n",
      "4_S4_R1_001.fastq.gz\r\n",
      "4_S4_R2_001.fastq.gz\r\n",
      "5_S5_R1_001.fastq.gz\r\n",
      "5_S5_R2_001.fastq.gz\r\n",
      "6_S6_R1_001.fastq.gz\r\n",
      "6_S6_R2_001.fastq.gz\r\n",
      "7_S7_R1_001.fastq.gz\r\n",
      "7_S7_R2_001.fastq.gz\r\n",
      "8_S8_R1_001.fastq.gz\r\n",
      "8_S8_R2_001.fastq.gz\r\n",
      "9_S9_R1_001.fastq.gz\r\n",
      "9_S9_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "#List fastq.gz files to be quality controlled\n",
    "!ls -1 ../../data/fastp_Kaiju_processing/input/*fastq.gz |xargs basename"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The first script will output the fastp commands that will be executed as a dry run. The second will execute the commands. Edit max_cores according to the number of cores you wish to use in .sh file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2021, March 22). GNU Parallel 20210322 ('2002-01-06').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.4628277\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 73 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "fastp -h 10_S10_R1_001.fastq.html -j 10_S10_R1_001.fastq.json -i 10_S10_R1_001.fastq.gz -I 10_S10_R2_001.fastq.gz --out1 10_S10_R1_001.fastq.out.gz --out2 10_S10_R2_001.fastq.out.gz\n",
      "fastp -h 11_S11_R1_001.fastq.html -j 11_S11_R1_001.fastq.json -i 11_S11_R1_001.fastq.gz -I 11_S11_R2_001.fastq.gz --out1 11_S11_R1_001.fastq.out.gz --out2 11_S11_R2_001.fastq.out.gz\n",
      "fastp -h 12_S12_R1_001.fastq.html -j 12_S12_R1_001.fastq.json -i 12_S12_R1_001.fastq.gz -I 12_S12_R2_001.fastq.gz --out1 12_S12_R1_001.fastq.out.gz --out2 12_S12_R2_001.fastq.out.gz\n",
      "fastp -h 13_S13_R1_001.fastq.html -j 13_S13_R1_001.fastq.json -i 13_S13_R1_001.fastq.gz -I 13_S13_R2_001.fastq.gz --out1 13_S13_R1_001.fastq.out.gz --out2 13_S13_R2_001.fastq.out.gz\n",
      "fastp -h 14_S14_R1_001.fastq.html -j 14_S14_R1_001.fastq.json -i 14_S14_R1_001.fastq.gz -I 14_S14_R2_001.fastq.gz --out1 14_S14_R1_001.fastq.out.gz --out2 14_S14_R2_001.fastq.out.gz\n",
      "fastp -h 15_S15_R1_001.fastq.html -j 15_S15_R1_001.fastq.json -i 15_S15_R1_001.fastq.gz -I 15_S15_R2_001.fastq.gz --out1 15_S15_R1_001.fastq.out.gz --out2 15_S15_R2_001.fastq.out.gz\n",
      "fastp -h 16_S16_R1_001.fastq.html -j 16_S16_R1_001.fastq.json -i 16_S16_R1_001.fastq.gz -I 16_S16_R2_001.fastq.gz --out1 16_S16_R1_001.fastq.out.gz --out2 16_S16_R2_001.fastq.out.gz\n",
      "fastp -h 17_S17_R1_001.fastq.html -j 17_S17_R1_001.fastq.json -i 17_S17_R1_001.fastq.gz -I 17_S17_R2_001.fastq.gz --out1 17_S17_R1_001.fastq.out.gz --out2 17_S17_R2_001.fastq.out.gz\n",
      "fastp -h 18_S18_R1_001.fastq.html -j 18_S18_R1_001.fastq.json -i 18_S18_R1_001.fastq.gz -I 18_S18_R2_001.fastq.gz --out1 18_S18_R1_001.fastq.out.gz --out2 18_S18_R2_001.fastq.out.gz\n",
      "fastp -h 19_S19_R1_001.fastq.html -j 19_S19_R1_001.fastq.json -i 19_S19_R1_001.fastq.gz -I 19_S19_R2_001.fastq.gz --out1 19_S19_R1_001.fastq.out.gz --out2 19_S19_R2_001.fastq.out.gz\n",
      "fastp -h 1_S1_R1_001.fastq.html -j 1_S1_R1_001.fastq.json -i 1_S1_R1_001.fastq.gz -I 1_S1_R2_001.fastq.gz --out1 1_S1_R1_001.fastq.out.gz --out2 1_S1_R2_001.fastq.out.gz\n",
      "fastp -h 20_S20_R1_001.fastq.html -j 20_S20_R1_001.fastq.json -i 20_S20_R1_001.fastq.gz -I 20_S20_R2_001.fastq.gz --out1 20_S20_R1_001.fastq.out.gz --out2 20_S20_R2_001.fastq.out.gz\n",
      "fastp -h 21_S21_R1_001.fastq.html -j 21_S21_R1_001.fastq.json -i 21_S21_R1_001.fastq.gz -I 21_S21_R2_001.fastq.gz --out1 21_S21_R1_001.fastq.out.gz --out2 21_S21_R2_001.fastq.out.gz\n",
      "fastp -h 22_S22_R1_001.fastq.html -j 22_S22_R1_001.fastq.json -i 22_S22_R1_001.fastq.gz -I 22_S22_R2_001.fastq.gz --out1 22_S22_R1_001.fastq.out.gz --out2 22_S22_R2_001.fastq.out.gz\n",
      "fastp -h 2_S2_R1_001.fastq.html -j 2_S2_R1_001.fastq.json -i 2_S2_R1_001.fastq.gz -I 2_S2_R2_001.fastq.gz --out1 2_S2_R1_001.fastq.out.gz --out2 2_S2_R2_001.fastq.out.gz\n",
      "fastp -h 3_S3_R1_001.fastq.html -j 3_S3_R1_001.fastq.json -i 3_S3_R1_001.fastq.gz -I 3_S3_R2_001.fastq.gz --out1 3_S3_R1_001.fastq.out.gz --out2 3_S3_R2_001.fastq.out.gz\n",
      "fastp -h 4_S4_R1_001.fastq.html -j 4_S4_R1_001.fastq.json -i 4_S4_R1_001.fastq.gz -I 4_S4_R2_001.fastq.gz --out1 4_S4_R1_001.fastq.out.gz --out2 4_S4_R2_001.fastq.out.gz\n",
      "fastp -h 5_S5_R1_001.fastq.html -j 5_S5_R1_001.fastq.json -i 5_S5_R1_001.fastq.gz -I 5_S5_R2_001.fastq.gz --out1 5_S5_R1_001.fastq.out.gz --out2 5_S5_R2_001.fastq.out.gz\n",
      "fastp -h 6_S6_R1_001.fastq.html -j 6_S6_R1_001.fastq.json -i 6_S6_R1_001.fastq.gz -I 6_S6_R2_001.fastq.gz --out1 6_S6_R1_001.fastq.out.gz --out2 6_S6_R2_001.fastq.out.gz\n",
      "fastp -h 7_S7_R1_001.fastq.html -j 7_S7_R1_001.fastq.json -i 7_S7_R1_001.fastq.gz -I 7_S7_R2_001.fastq.gz --out1 7_S7_R1_001.fastq.out.gz --out2 7_S7_R2_001.fastq.out.gz\n",
      "fastp -h 8_S8_R1_001.fastq.html -j 8_S8_R1_001.fastq.json -i 8_S8_R1_001.fastq.gz -I 8_S8_R2_001.fastq.gz --out1 8_S8_R1_001.fastq.out.gz --out2 8_S8_R2_001.fastq.out.gz\n",
      "fastp -h 9_S9_R1_001.fastq.html -j 9_S9_R1_001.fastq.json -i 9_S9_R1_001.fastq.gz -I 9_S9_R2_001.fastq.gz --out1 9_S9_R1_001.fastq.out.gz --out2 9_S9_R2_001.fastq.out.gz\n"
     ]
    }
   ],
   "source": [
    "!./fastp_dryrun.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2021, March 22). GNU Parallel 20210322 ('2002-01-06').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.4628277\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 74 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 5054869\n",
      "total bases: 741560849\n",
      "Q20 bases: 723007733(97.4981%)\n",
      "Q30 bases: 697286398(94.0296%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 5054869\n",
      "total bases: 741574756\n",
      "Q20 bases: 708574414(95.55%)\n",
      "Q30 bases: 670283850(90.3866%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 5014080\n",
      "total bases: 735590840\n",
      "Q20 bases: 718291750(97.6483%)\n",
      "Q30 bases: 693087132(94.2218%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 5014080\n",
      "total bases: 735501768\n",
      "Q20 bases: 704967032(95.8484%)\n",
      "Q30 bases: 667420065(90.7435%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 10028160\n",
      "reads failed due to low quality: 72888\n",
      "reads failed due to too many N: 8690\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 30122\n",
      "bases trimmed due to adapters: 977831\n",
      "\n",
      "Duplication rate: 16.6022%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 263\n",
      "\n",
      "JSON report: 1_S1_R1_001.fastq.json\n",
      "HTML report: 1_S1_R1_001.fastq.html\n",
      "\n",
      "fastp -h 1_S1_R1_001.fastq.html -j 1_S1_R1_001.fastq.json -i 1_S1_R1_001.fastq.gz -I 1_S1_R2_001.fastq.gz --out1 1_S1_R1_001.fastq.out.gz --out2 1_S1_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 169 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 6292688\n",
      "total bases: 879517074\n",
      "Q20 bases: 858240587(97.5809%)\n",
      "Q30 bases: 830466350(94.423%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6292688\n",
      "total bases: 879568330\n",
      "Q20 bases: 845465838(96.1228%)\n",
      "Q30 bases: 805838812(91.6175%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 6227099\n",
      "total bases: 871938594\n",
      "Q20 bases: 852404519(97.7597%)\n",
      "Q30 bases: 825231626(94.6433%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 6227099\n",
      "total bases: 871598305\n",
      "Q20 bases: 840421195(96.423%)\n",
      "Q30 bases: 801652970(91.975%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 12454198\n",
      "reads failed due to low quality: 91672\n",
      "reads failed due to too many N: 39506\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 67824\n",
      "bases trimmed due to adapters: 895093\n",
      "\n",
      "Duplication rate: 18.7978%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 64\n",
      "\n",
      "JSON report: 12_S12_R1_001.fastq.json\n",
      "HTML report: 12_S12_R1_001.fastq.html\n",
      "\n",
      "fastp -h 12_S12_R1_001.fastq.html -j 12_S12_R1_001.fastq.json -i 12_S12_R1_001.fastq.gz -I 12_S12_R2_001.fastq.gz --out1 12_S12_R1_001.fastq.out.gz --out2 12_S12_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 190 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 5873374\n",
      "total bases: 858110155\n",
      "Q20 bases: 838023906(97.6592%)\n",
      "Q30 bases: 808937999(94.2697%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 5873374\n",
      "total bases: 858163728\n",
      "Q20 bases: 822497185(95.8439%)\n",
      "Q30 bases: 779141271(90.7917%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 5835953\n",
      "total bases: 852956917\n",
      "Q20 bases: 833957638(97.7725%)\n",
      "Q30 bases: 805332534(94.4166%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 5835953\n",
      "total bases: 852872183\n",
      "Q20 bases: 819329185(96.0671%)\n",
      "Q30 bases: 776628710(91.0604%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 11671906\n",
      "reads failed due to low quality: 63396\n",
      "reads failed due to too many N: 11446\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 29946\n",
      "bases trimmed due to adapters: 694813\n",
      "\n",
      "Duplication rate: 17.0415%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 265\n",
      "\n",
      "JSON report: 5_S5_R1_001.fastq.json\n",
      "HTML report: 5_S5_R1_001.fastq.html\n",
      "\n",
      "fastp -h 5_S5_R1_001.fastq.html -j 5_S5_R1_001.fastq.json -i 5_S5_R1_001.fastq.gz -I 5_S5_R2_001.fastq.gz --out1 5_S5_R1_001.fastq.out.gz --out2 5_S5_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 192 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 6074230\n",
      "total bases: 893837766\n",
      "Q20 bases: 871195714(97.4669%)\n",
      "Q30 bases: 841534002(94.1484%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6074230\n",
      "total bases: 893810470\n",
      "Q20 bases: 855915743(95.7603%)\n",
      "Q30 bases: 812343923(90.8855%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 6022515\n",
      "total bases: 886102651\n",
      "Q20 bases: 865243796(97.646%)\n",
      "Q30 bases: 836236698(94.3724%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 6022515\n",
      "total bases: 885965758\n",
      "Q20 bases: 851066119(96.0608%)\n",
      "Q30 bases: 808391226(91.2441%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 12045030\n",
      "reads failed due to low quality: 93050\n",
      "reads failed due to too many N: 10380\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 39680\n",
      "bases trimmed due to adapters: 1435062\n",
      "\n",
      "Duplication rate: 18.0868%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 263\n",
      "\n",
      "JSON report: 2_S2_R1_001.fastq.json\n",
      "HTML report: 2_S2_R1_001.fastq.html\n",
      "\n",
      "fastp -h 2_S2_R1_001.fastq.html -j 2_S2_R1_001.fastq.json -i 2_S2_R1_001.fastq.gz -I 2_S2_R2_001.fastq.gz --out1 2_S2_R1_001.fastq.out.gz --out2 2_S2_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 199 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 6228988\n",
      "total bases: 913314039\n",
      "Q20 bases: 891317407(97.5916%)\n",
      "Q30 bases: 862066577(94.3888%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6228988\n",
      "total bases: 913309125\n",
      "Q20 bases: 875194031(95.8267%)\n",
      "Q30 bases: 831216537(91.0115%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 6179371\n",
      "total bases: 906215658\n",
      "Q20 bases: 885799154(97.7471%)\n",
      "Q30 bases: 857154525(94.5862%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 6179371\n",
      "total bases: 906081784\n",
      "Q20 bases: 870703742(96.0955%)\n",
      "Q30 bases: 827562238(91.3342%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 12358742\n",
      "reads failed due to low quality: 88662\n",
      "reads failed due to too many N: 10572\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 31744\n",
      "bases trimmed due to adapters: 815175\n",
      "\n",
      "Duplication rate: 18.2575%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 265\n",
      "\n",
      "JSON report: 7_S7_R1_001.fastq.json\n",
      "HTML report: 7_S7_R1_001.fastq.html\n",
      "\n",
      "fastp -h 7_S7_R1_001.fastq.html -j 7_S7_R1_001.fastq.json -i 7_S7_R1_001.fastq.gz -I 7_S7_R2_001.fastq.gz --out1 7_S7_R1_001.fastq.out.gz --out2 7_S7_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 202 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 6584222\n",
      "total bases: 963343077\n",
      "Q20 bases: 941235693(97.7051%)\n",
      "Q30 bases: 909838639(94.446%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6584222\n",
      "total bases: 963423805\n",
      "Q20 bases: 920755084(95.5711%)\n",
      "Q30 bases: 871049786(90.4119%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 6535509\n",
      "total bases: 956510479\n",
      "Q20 bases: 935761078(97.8307%)\n",
      "Q30 bases: 904957497(94.6103%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 6535509\n",
      "total bases: 956425724\n",
      "Q20 bases: 916647747(95.841%)\n",
      "Q30 bases: 867826675(90.7364%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 13071018\n",
      "reads failed due to low quality: 85314\n",
      "reads failed due to too many N: 12112\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 32306\n",
      "bases trimmed due to adapters: 774922\n",
      "\n",
      "Duplication rate: 19.7947%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 262\n",
      "\n",
      "JSON report: 3_S3_R1_001.fastq.json\n",
      "HTML report: 3_S3_R1_001.fastq.html\n",
      "\n",
      "fastp -h 3_S3_R1_001.fastq.html -j 3_S3_R1_001.fastq.json -i 3_S3_R1_001.fastq.gz -I 3_S3_R2_001.fastq.gz --out1 3_S3_R1_001.fastq.out.gz --out2 3_S3_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 210 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 6845208\n",
      "total bases: 1001669923\n",
      "Q20 bases: 978924854(97.7293%)\n",
      "Q30 bases: 946754483(94.5176%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6845208\n",
      "total bases: 1001732302\n",
      "Q20 bases: 961846035(96.0183%)\n",
      "Q30 bases: 913021090(91.1442%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 6801215\n",
      "total bases: 995451061\n",
      "Q20 bases: 974055280(97.8506%)\n",
      "Q30 bases: 942410544(94.6717%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 6801215\n",
      "total bases: 995334389\n",
      "Q20 bases: 957915403(96.2406%)\n",
      "Q30 bases: 909840967(91.4106%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 13602430\n",
      "reads failed due to low quality: 77422\n",
      "reads failed due to too many N: 10564\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 32486\n",
      "bases trimmed due to adapters: 799777\n",
      "\n",
      "Duplication rate: 17.5496%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 264\n",
      "\n",
      "JSON report: 19_S19_R1_001.fastq.json\n",
      "HTML report: 19_S19_R1_001.fastq.html\n",
      "\n",
      "fastp -h 19_S19_R1_001.fastq.html -j 19_S19_R1_001.fastq.json -i 19_S19_R1_001.fastq.gz -I 19_S19_R2_001.fastq.gz --out1 19_S19_R1_001.fastq.out.gz --out2 19_S19_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 214 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 7280079\n",
      "total bases: 1033323307\n",
      "Q20 bases: 1008906915(97.6371%)\n",
      "Q30 bases: 975659915(94.4196%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7280079\n",
      "total bases: 1033446183\n",
      "Q20 bases: 993300675(96.1154%)\n",
      "Q30 bases: 945799722(91.519%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7216219\n",
      "total bases: 1025762948\n",
      "Q20 bases: 1003025813(97.7834%)\n",
      "Q30 bases: 970398180(94.6026%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7216219\n",
      "total bases: 1025557556\n",
      "Q20 bases: 988413177(96.3781%)\n",
      "Q30 bases: 941808145(91.8338%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14432438\n",
      "reads failed due to low quality: 91422\n",
      "reads failed due to too many N: 36298\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 64828\n",
      "bases trimmed due to adapters: 913019\n",
      "\n",
      "Duplication rate: 20.3093%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 265\n",
      "\n",
      "JSON report: 8_S8_R1_001.fastq.json\n",
      "HTML report: 8_S8_R1_001.fastq.html\n",
      "\n",
      "fastp -h 8_S8_R1_001.fastq.html -j 8_S8_R1_001.fastq.json -i 8_S8_R1_001.fastq.gz -I 8_S8_R2_001.fastq.gz --out1 8_S8_R1_001.fastq.out.gz --out2 8_S8_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 215 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7080173\n",
      "total bases: 1032207488\n",
      "Q20 bases: 1008221108(97.6762%)\n",
      "Q30 bases: 972666823(94.2317%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7080173\n",
      "total bases: 1032346723\n",
      "Q20 bases: 988439046(95.7468%)\n",
      "Q30 bases: 934604251(90.532%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7041035\n",
      "total bases: 1026788658\n",
      "Q20 bases: 1003906419(97.7715%)\n",
      "Q30 bases: 968845958(94.3569%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7041035\n",
      "total bases: 1026717753\n",
      "Q20 bases: 985143539(95.9508%)\n",
      "Q30 bases: 932026659(90.7773%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14082070\n",
      "reads failed due to low quality: 66140\n",
      "reads failed due to too many N: 12136\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 36654\n",
      "bases trimmed due to adapters: 854275\n",
      "\n",
      "Duplication rate: 19.1509%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 265\n",
      "\n",
      "JSON report: 21_S21_R1_001.fastq.json\n",
      "HTML report: 21_S21_R1_001.fastq.html\n",
      "\n",
      "fastp -h 21_S21_R1_001.fastq.html -j 21_S21_R1_001.fastq.json -i 21_S21_R1_001.fastq.gz -I 21_S21_R2_001.fastq.gz --out1 21_S21_R1_001.fastq.out.gz --out2 21_S21_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 218 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7311119\n",
      "total bases: 1041500607\n",
      "Q20 bases: 1016920369(97.6399%)\n",
      "Q30 bases: 981945260(94.2818%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7311119\n",
      "total bases: 1041665335\n",
      "Q20 bases: 998602953(95.866%)\n",
      "Q30 bases: 947383273(90.9489%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7249449\n",
      "total bases: 1033955628\n",
      "Q20 bases: 1010995664(97.7794%)\n",
      "Q30 bases: 976665732(94.4592%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7249449\n",
      "total bases: 1033755238\n",
      "Q20 bases: 993855696(96.1403%)\n",
      "Q30 bases: 943575550(91.2765%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14498898\n",
      "reads failed due to low quality: 90102\n",
      "reads failed due to too many N: 33238\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 63848\n",
      "bases trimmed due to adapters: 1138942\n",
      "\n",
      "Duplication rate: 19.1133%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 266\n",
      "\n",
      "JSON report: 4_S4_R1_001.fastq.json\n",
      "HTML report: 4_S4_R1_001.fastq.html\n",
      "\n",
      "fastp -h 4_S4_R1_001.fastq.html -j 4_S4_R1_001.fastq.json -i 4_S4_R1_001.fastq.gz -I 4_S4_R2_001.fastq.gz --out1 4_S4_R1_001.fastq.out.gz --out2 4_S4_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 218 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7290173\n",
      "total bases: 1047893921\n",
      "Q20 bases: 1024165698(97.7356%)\n",
      "Q30 bases: 991399388(94.6088%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7290173\n",
      "total bases: 1048007632\n",
      "Q20 bases: 1001930135(95.6033%)\n",
      "Q30 bases: 948602225(90.5148%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7229047\n",
      "total bases: 1039902831\n",
      "Q20 bases: 1017869187(97.8812%)\n",
      "Q30 bases: 985786650(94.796%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7229047\n",
      "total bases: 1039724742\n",
      "Q20 bases: 996920530(95.8831%)\n",
      "Q30 bases: 944597327(90.8507%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14458094\n",
      "reads failed due to low quality: 99428\n",
      "reads failed due to too many N: 22824\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 51114\n",
      "bases trimmed due to adapters: 869902\n",
      "\n",
      "Duplication rate: 21.2479%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 263\n",
      "\n",
      "JSON report: 6_S6_R1_001.fastq.json\n",
      "HTML report: 6_S6_R1_001.fastq.html\n",
      "\n",
      "fastp -h 6_S6_R1_001.fastq.html -j 6_S6_R1_001.fastq.json -i 6_S6_R1_001.fastq.gz -I 6_S6_R2_001.fastq.gz --out1 6_S6_R1_001.fastq.out.gz --out2 6_S6_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 219 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7082859\n",
      "total bases: 1042042935\n",
      "Q20 bases: 1017843474(97.6777%)\n",
      "Q30 bases: 984803017(94.507%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7082859\n",
      "total bases: 1042078893\n",
      "Q20 bases: 999091232(95.8748%)\n",
      "Q30 bases: 948937178(91.0619%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7030728\n",
      "total bases: 1034556393\n",
      "Q20 bases: 1011964497(97.8163%)\n",
      "Q30 bases: 979568536(94.6849%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7030728\n",
      "total bases: 1034460431\n",
      "Q20 bases: 994425953(96.1299%)\n",
      "Q30 bases: 945185832(91.3699%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14061456\n",
      "reads failed due to low quality: 94216\n",
      "reads failed due to too many N: 10046\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 28798\n",
      "bases trimmed due to adapters: 749361\n",
      "\n",
      "Duplication rate: 19.8962%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 264\n",
      "\n",
      "JSON report: 13_S13_R1_001.fastq.json\n",
      "HTML report: 13_S13_R1_001.fastq.html\n",
      "\n",
      "fastp -h 13_S13_R1_001.fastq.html -j 13_S13_R1_001.fastq.json -i 13_S13_R1_001.fastq.gz -I 13_S13_R2_001.fastq.gz --out1 13_S13_R1_001.fastq.out.gz --out2 13_S13_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 220 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7083626\n",
      "total bases: 1042030108\n",
      "Q20 bases: 1017430526(97.6393%)\n",
      "Q30 bases: 983358750(94.3695%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7083626\n",
      "total bases: 1042095618\n",
      "Q20 bases: 999085446(95.8727%)\n",
      "Q30 bases: 947542530(90.9266%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7032599\n",
      "total bases: 1034646131\n",
      "Q20 bases: 1011610079(97.7735%)\n",
      "Q30 bases: 978165579(94.5411%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7032599\n",
      "total bases: 1034567144\n",
      "Q20 bases: 994481005(96.1253%)\n",
      "Q30 bases: 943831624(91.2296%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14065198\n",
      "reads failed due to low quality: 90842\n",
      "reads failed due to too many N: 11212\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 34920\n",
      "bases trimmed due to adapters: 1041051\n",
      "\n",
      "Duplication rate: 19.2032%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 264\n",
      "\n",
      "JSON report: 16_S16_R1_001.fastq.json\n",
      "HTML report: 16_S16_R1_001.fastq.html\n",
      "\n",
      "fastp -h 16_S16_R1_001.fastq.html -j 16_S16_R1_001.fastq.json -i 16_S16_R1_001.fastq.gz -I 16_S16_R2_001.fastq.gz --out1 16_S16_R1_001.fastq.out.gz --out2 16_S16_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 221 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7242019\n",
      "total bases: 1064868146\n",
      "Q20 bases: 1039351242(97.6037%)\n",
      "Q30 bases: 1004895036(94.368%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7242019\n",
      "total bases: 1064858789\n",
      "Q20 bases: 1023634331(96.1286%)\n",
      "Q30 bases: 974158576(91.4824%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7191645\n",
      "total bases: 1057525892\n",
      "Q20 bases: 1033688848(97.746%)\n",
      "Q30 bases: 999858442(94.5469%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7191645\n",
      "total bases: 1057368534\n",
      "Q20 bases: 1018925346(96.3643%)\n",
      "Q30 bases: 970308227(91.7663%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14383290\n",
      "reads failed due to low quality: 91244\n",
      "reads failed due to too many N: 9504\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 33096\n",
      "bases trimmed due to adapters: 986311\n",
      "\n",
      "Duplication rate: 20.0147%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 264\n",
      "\n",
      "JSON report: 10_S10_R1_001.fastq.json\n",
      "HTML report: 10_S10_R1_001.fastq.html\n",
      "\n",
      "fastp -h 10_S10_R1_001.fastq.html -j 10_S10_R1_001.fastq.json -i 10_S10_R1_001.fastq.gz -I 10_S10_R2_001.fastq.gz --out1 10_S10_R1_001.fastq.out.gz --out2 10_S10_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 221 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7902499\n",
      "total bases: 1113430871\n",
      "Q20 bases: 1087702663(97.6893%)\n",
      "Q30 bases: 1052931727(94.5664%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7902499\n",
      "total bases: 1113552612\n",
      "Q20 bases: 1073218143(96.3779%)\n",
      "Q30 bases: 1023841516(91.9437%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7834755\n",
      "total bases: 1105412027\n",
      "Q20 bases: 1081561940(97.8424%)\n",
      "Q30 bases: 1047421275(94.7539%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7834755\n",
      "total bases: 1105088462\n",
      "Q20 bases: 1067775766(96.6236%)\n",
      "Q30 bases: 1019289154(92.236%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 15669510\n",
      "reads failed due to low quality: 96338\n",
      "reads failed due to too many N: 39150\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 75586\n",
      "bases trimmed due to adapters: 1167599\n",
      "\n",
      "Duplication rate: 22.38%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 263\n",
      "\n",
      "JSON report: 17_S17_R1_001.fastq.json\n",
      "HTML report: 17_S17_R1_001.fastq.html\n",
      "\n",
      "fastp -h 17_S17_R1_001.fastq.html -j 17_S17_R1_001.fastq.json -i 17_S17_R1_001.fastq.gz -I 17_S17_R2_001.fastq.gz --out1 17_S17_R1_001.fastq.out.gz --out2 17_S17_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 222 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 7723028\n",
      "total bases: 1120466178\n",
      "Q20 bases: 1093895734(97.6286%)\n",
      "Q30 bases: 1058139890(94.4375%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7723028\n",
      "total bases: 1120474180\n",
      "Q20 bases: 1077516760(96.1661%)\n",
      "Q30 bases: 1025823883(91.5527%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7664435\n",
      "total bases: 1112319005\n",
      "Q20 bases: 1087651849(97.7824%)\n",
      "Q30 bases: 1052568032(94.6283%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 7664435\n",
      "total bases: 1112103394\n",
      "Q20 bases: 1072170462(96.4092%)\n",
      "Q30 bases: 1021385439(91.8427%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 15328870\n",
      "reads failed due to low quality: 99444\n",
      "reads failed due to too many N: 17742\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 50412\n",
      "bases trimmed due to adapters: 1296175\n",
      "\n",
      "Duplication rate: 18.957%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 264\n",
      "\n",
      "JSON report: 18_S18_R1_001.fastq.json\n",
      "HTML report: 18_S18_R1_001.fastq.html\n",
      "\n",
      "fastp -h 18_S18_R1_001.fastq.html -j 18_S18_R1_001.fastq.json -i 18_S18_R1_001.fastq.gz -I 18_S18_R2_001.fastq.gz --out1 18_S18_R1_001.fastq.out.gz --out2 18_S18_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 224 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 8107184\n",
      "total bases: 1162899236\n",
      "Q20 bases: 1137155484(97.7862%)\n",
      "Q30 bases: 1100744061(94.6552%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 8107184\n",
      "total bases: 1163016152\n",
      "Q20 bases: 1123896208(96.6363%)\n",
      "Q30 bases: 1072854140(92.2476%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 8055744\n",
      "total bases: 1156331868\n",
      "Q20 bases: 1132140524(97.9079%)\n",
      "Q30 bases: 1096255857(94.8046%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 8055744\n",
      "total bases: 1156106809\n",
      "Q20 bases: 1119415287(96.8263%)\n",
      "Q30 bases: 1069095356(92.4738%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 16111488\n",
      "reads failed due to low quality: 79054\n",
      "reads failed due to too many N: 23826\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 58066\n",
      "bases trimmed due to adapters: 1120213\n",
      "\n",
      "Duplication rate: 21.2503%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 265\n",
      "\n",
      "JSON report: 20_S20_R1_001.fastq.json\n",
      "HTML report: 20_S20_R1_001.fastq.html\n",
      "\n",
      "fastp -h 20_S20_R1_001.fastq.html -j 20_S20_R1_001.fastq.json -i 20_S20_R1_001.fastq.gz -I 20_S20_R2_001.fastq.gz --out1 20_S20_R1_001.fastq.out.gz --out2 20_S20_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 226 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 8064502\n",
      "total bases: 1185929539\n",
      "Q20 bases: 1159145636(97.7415%)\n",
      "Q30 bases: 1123166305(94.7077%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 8064502\n",
      "total bases: 1185905766\n",
      "Q20 bases: 1141591491(96.2633%)\n",
      "Q30 bases: 1087721485(91.7207%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 8008510\n",
      "total bases: 1177940979\n",
      "Q20 bases: 1153032487(97.8854%)\n",
      "Q30 bases: 1117725148(94.888%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 8008510\n",
      "total bases: 1177772504\n",
      "Q20 bases: 1136369609(96.4846%)\n",
      "Q30 bases: 1083387025(91.9861%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 16017020\n",
      "reads failed due to low quality: 100944\n",
      "reads failed due to too many N: 11040\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 32754\n",
      "bases trimmed due to adapters: 808130\n",
      "\n",
      "Duplication rate: 20.3628%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 264\n",
      "\n",
      "JSON report: 9_S9_R1_001.fastq.json\n",
      "HTML report: 9_S9_R1_001.fastq.html\n",
      "\n",
      "fastp -h 9_S9_R1_001.fastq.html -j 9_S9_R1_001.fastq.json -i 9_S9_R1_001.fastq.gz -I 9_S9_R2_001.fastq.gz --out1 9_S9_R1_001.fastq.out.gz --out2 9_S9_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 229 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 9267570\n",
      "total bases: 1366421971\n",
      "Q20 bases: 1334183014(97.6406%)\n",
      "Q30 bases: 1291182620(94.4937%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 9267570\n",
      "total bases: 1366410455\n",
      "Q20 bases: 1316838467(96.3721%)\n",
      "Q30 bases: 1255855363(91.9091%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 9203705\n",
      "total bases: 1356627524\n",
      "Q20 bases: 1326830239(97.8036%)\n",
      "Q30 bases: 1284651734(94.6945%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 9203705\n",
      "total bases: 1356436560\n",
      "Q20 bases: 1310437740(96.6088%)\n",
      "Q30 bases: 1250512409(92.191%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 18407410\n",
      "reads failed due to low quality: 117318\n",
      "reads failed due to too many N: 10412\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 51750\n",
      "bases trimmed due to adapters: 2010727\n",
      "\n",
      "Duplication rate: 21.2525%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 263\n",
      "\n",
      "JSON report: 22_S22_R1_001.fastq.json\n",
      "HTML report: 22_S22_R1_001.fastq.html\n",
      "\n",
      "fastp -h 22_S22_R1_001.fastq.html -j 22_S22_R1_001.fastq.json -i 22_S22_R1_001.fastq.gz -I 22_S22_R2_001.fastq.gz --out1 22_S22_R1_001.fastq.out.gz --out2 22_S22_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 238 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 10315442\n",
      "total bases: 1452410678\n",
      "Q20 bases: 1421544745(97.8748%)\n",
      "Q30 bases: 1377535609(94.8448%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 10315442\n",
      "total bases: 1452694842\n",
      "Q20 bases: 1402800380(96.5654%)\n",
      "Q30 bases: 1338771719(92.1578%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 10243208\n",
      "total bases: 1444181658\n",
      "Q20 bases: 1415296412(97.9999%)\n",
      "Q30 bases: 1371927523(94.9969%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 10243208\n",
      "total bases: 1443827664\n",
      "Q20 bases: 1397087280(96.7627%)\n",
      "Q30 bases: 1333976099(92.3916%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 20486416\n",
      "reads failed due to low quality: 98100\n",
      "reads failed due to too many N: 46368\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 90806\n",
      "bases trimmed due to adapters: 1294636\n",
      "\n",
      "Duplication rate: 21.26%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 264\n",
      "\n",
      "JSON report: 15_S15_R1_001.fastq.json\n",
      "HTML report: 15_S15_R1_001.fastq.html\n",
      "\n",
      "fastp -h 15_S15_R1_001.fastq.html -j 15_S15_R1_001.fastq.json -i 15_S15_R1_001.fastq.gz -I 15_S15_R2_001.fastq.gz --out1 15_S15_R1_001.fastq.out.gz --out2 15_S15_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 239 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 10214859\n",
      "total bases: 1466443339\n",
      "Q20 bases: 1430346991(97.5385%)\n",
      "Q30 bases: 1385302473(94.4668%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 10214859\n",
      "total bases: 1466357680\n",
      "Q20 bases: 1411567644(96.2635%)\n",
      "Q30 bases: 1346451912(91.8229%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 10116366\n",
      "total bases: 1453096966\n",
      "Q20 bases: 1420342151(97.7459%)\n",
      "Q30 bases: 1376353018(94.7186%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 10116366\n",
      "total bases: 1452561351\n",
      "Q20 bases: 1402514610(96.5546%)\n",
      "Q30 bases: 1338778939(92.1668%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 20232732\n",
      "reads failed due to low quality: 161892\n",
      "reads failed due to too many N: 35094\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 92496\n",
      "bases trimmed due to adapters: 2267651\n",
      "\n",
      "Duplication rate: 22.6016%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 266\n",
      "\n",
      "JSON report: 11_S11_R1_001.fastq.json\n",
      "HTML report: 11_S11_R1_001.fastq.html\n",
      "\n",
      "fastp -h 11_S11_R1_001.fastq.html -j 11_S11_R1_001.fastq.json -i 11_S11_R1_001.fastq.gz -I 11_S11_R2_001.fastq.gz --out1 11_S11_R1_001.fastq.out.gz --out2 11_S11_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 242 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 10987024\n",
      "total bases: 1558857731\n",
      "Q20 bases: 1525417085(97.8548%)\n",
      "Q30 bases: 1476191702(94.697%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 10987024\n",
      "total bases: 1559220339\n",
      "Q20 bases: 1501349421(96.2885%)\n",
      "Q30 bases: 1426537612(91.4904%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 10920434\n",
      "total bases: 1550881266\n",
      "Q20 bases: 1519368806(97.9681%)\n",
      "Q30 bases: 1470789149(94.8357%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 10920434\n",
      "total bases: 1550621190\n",
      "Q20 bases: 1495817544(96.4657%)\n",
      "Q30 bases: 1421909048(91.6993%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 21840868\n",
      "reads failed due to low quality: 93930\n",
      "reads failed due to too many N: 39250\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 93794\n",
      "bases trimmed due to adapters: 1640808\n",
      "\n",
      "Duplication rate: 23.7793%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 263\n",
      "\n",
      "JSON report: 14_S14_R1_001.fastq.json\n",
      "HTML report: 14_S14_R1_001.fastq.html\n",
      "\n",
      "fastp -h 14_S14_R1_001.fastq.html -j 14_S14_R1_001.fastq.json -i 14_S14_R1_001.fastq.gz -I 14_S14_R2_001.fastq.gz --out1 14_S14_R1_001.fastq.out.gz --out2 14_S14_R2_001.fastq.out.gz \n",
      "fastp v0.20.1, time used: 245 seconds\n"
     ]
    }
   ],
   "source": [
    "#This script also moves the output into the output folder\n",
    "!./fastp.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Kaiju Taxonomic Analysis:\n",
    "Run Kaiju on the quality controlled reads (fastq.out.gz)  \n",
    "This example uses the nr+euk database from NCBI. \n",
    "Creates list of R1 and R2 files.\n",
    "Runs Kaiju with all samples, this may take a while.\n",
    "Change -z option in .sh file based on number of cores you wish to use\n",
    "This will output a kaiju.out file with information about the taxonomy for each sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "!./kaiju.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Kaiju2Table:\n",
    "This will convert the kaiju.out files to readable .tsv tables\n",
    "It is currently set to list all the way to species, but it can be changed to whic ever level you desire\n",
    "If changed, you will have to edit modify_otu.py according to .tsv file\n",
    "First file is also a dry run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2021, March 22). GNU Parallel 20210322 ('2002-01-06').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.4628277\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 79 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 10_S10_R1_001.fastq.tsv 10_S10_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 11_S11_R1_001.fastq.tsv 11_S11_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 12_S12_R1_001.fastq.tsv 12_S12_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 13_S13_R1_001.fastq.tsv 13_S13_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 14_S14_R1_001.fastq.tsv 14_S14_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 15_S15_R1_001.fastq.tsv 15_S15_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 16_S16_R1_001.fastq.tsv 16_S16_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 17_S17_R1_001.fastq.tsv 17_S17_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 18_S18_R1_001.fastq.tsv 18_S18_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 19_S19_R1_001.fastq.tsv 19_S19_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 1_S1_R1_001.fastq.tsv 1_S1_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 20_S20_R1_001.fastq.tsv 20_S20_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 21_S21_R1_001.fastq.tsv 21_S21_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 22_S22_R1_001.fastq.tsv 22_S22_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 2_S2_R1_001.fastq.tsv 2_S2_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 3_S3_R1_001.fastq.tsv 3_S3_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 4_S4_R1_001.fastq.tsv 4_S4_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 5_S5_R1_001.fastq.tsv 5_S5_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 6_S6_R1_001.fastq.tsv 6_S6_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 7_S7_R1_001.fastq.tsv 7_S7_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 8_S8_R1_001.fastq.tsv 8_S8_R1_001.fastq.out\n",
      "kaiju2table -t ../../input/nodes.dmp -n ../../input/names.dmp -r species -l domain,superkingdom,phylum,class,order,family,genus,species -o 9_S9_R1_001.fastq.tsv 9_S9_R1_001.fastq.out\n"
     ]
    }
   ],
   "source": [
    "!./kaiju_2_table_dryrun.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Academic tradition requires you to cite works you base your article on.\r\n",
      "If you use programs that use GNU Parallel to process data for an article in a\r\n",
      "scientific publication, please cite:\r\n",
      "\r\n",
      "  Tange, O. (2021, March 22). GNU Parallel 20210322 ('2002-01-06').\r\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.4628277\r\n",
      "\r\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\r\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\r\n",
      "\r\n",
      "More about funding GNU Parallel and the citation notice:\r\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\r\n",
      "\r\n",
      "To silence this citation notice: run 'parallel --citation' once.\r\n",
      "\r\n",
      "Come on: You have run parallel 80 times. Isn't it about time \r\n",
      "you run 'parallel --citation' once to silence the citation notice?\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!./kaiju_2_table.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Kaiju Table to OTU Table:\n",
    "This program, authored by Laura Sisk-Hackworth and modified by myself, combines all of the kaiju .tsv files and converts them into Bacteria, Archaea, Eukaryota, Viral, and metagenomic OTU tables."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['20_S20_R1_001.fastq.tsv', '7_S7_R1_001.fastq.tsv', '9_S9_R1_001.fastq.tsv', '3_S3_R1_001.fastq.tsv', '21_S21_R1_001.fastq.tsv', '22_S22_R1_001.fastq.tsv', '8_S8_R1_001.fastq.tsv', '6_S6_R1_001.fastq.tsv', '2_S2_R1_001.fastq.tsv', '5_S5_R1_001.fastq.tsv', '16_S16_R1_001.fastq.tsv', '17_S17_R1_001.fastq.tsv', '1_S1_R1_001.fastq.tsv', '15_S15_R1_001.fastq.tsv', '18_S18_R1_001.fastq.tsv', '19_S19_R1_001.fastq.tsv', '14_S14_R1_001.fastq.tsv', '11_S11_R1_001.fastq.tsv', '4_S4_R1_001.fastq.tsv', '10_S10_R1_001.fastq.tsv', '12_S12_R1_001.fastq.tsv', '13_S13_R1_001.fastq.tsv']\r\n"
     ]
    }
   ],
   "source": [
    "!python kaiju_to_otu.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Modify OTU Table:\n",
    "This program prepares the metagenomic OTU table for analysis in R.\n",
    "If you modified the kaiju_2_table settings, you will need to edit this as well. If -r in kaiju_2_table was changed, edit 'species' in modify_otu.py to what was specified after -r. If -l in kaiju_2_table was changed, drop columns accordingly in modify_otu.py.\n",
    "Outputs metagenomemodified.csv used in R analysis scripts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "!python modify_otu.py"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
